src/Aspire.Hosting.Azure.Sql/SqlServerScriptProvisioningResource.cs

```diff
@@ -0,0 +1,86 @@
+    private BicepValue<string>? _kind;
+    private BicepValue<string>? _azCliVersion;
+    private BicepValue<string>? _azPowerShellVersion;
+    private BicepValue<TimeSpan>? _retentionInterval;
+    private BicepList<ContainerAppEnvironmentVariable>? _environmentVariables;
+
+    public SqlServerScriptProvisioningResource(string bicepIdentifier) : base(bicepIdentifier)
+    {
+        RetentionInterval = TimeSpan.FromHours(1);
+    }
+
+    /// <summary>
+    /// Type of the script. e.g., AzurePowerShell, .
+    /// </summary>
+    public BicepValue<string> Kind
+    {
+        get { Initialize(); return _kind!; }
+        set { Initialize(); _kind!.Assign(value); }
+    }
+
+    /// <summary>
+    /// Script body.
+    /// </summary>
+    public BicepValue<string> ScriptContent
+    {
+        get { Initialize(); return _scriptContent!; }
+        set { Initialize(); _scriptContent!.Assign(value); }
+    }
+
+    /// <summary>
+    /// Azure CLI module version to be used.
+    /// </summary>
+    public BicepValue<string> AZCliVersion
+    {
+        get { Initialize(); return _azCliVersion!; }
+        set { Initialize(); _azCliVersion!.Assign(value); }
+    }
+
+    /// <summary>
+    /// Azure CLI module version to be used.
+    /// </summary>
+    public BicepValue<string> AZPowerShellVersion
+    {
+        get { Initialize(); return _azPowerShellVersion!; }
+        set { Initialize(); _azPowerShellVersion!.Assign(value); }
+    }
+
+    /// <summary>
+    /// Interval for which the service retains the script resource after it reaches a terminal state. Resource will be deleted when this duration expires. Duration is based on ISO 8601 pattern (for example P1D means one day).
+    /// </summary>
+    public BicepValue<TimeSpan> RetentionInterval
+    {
+        get { Initialize(); return _retentionInterval!; }
+        set { Initialize(); _retentionInterval!.Assign(value); }
+    }
+
+    public BicepList<ContainerAppEnvironmentVariable> EnvironmentVariables
+    {
+        get { Initialize(); return _environmentVariables!; }
+        set { Initialize(); _environmentVariables!.Assign(value); }
+    }
+
+    protected override void DefineProvisionableProperties()
+    {
+        base.DefineProvisionableProperties();
+
+        _kind = DefineProperty<string>(nameof(Kind), ["kind"], isRequired: true);
+        _scriptContent = DefineProperty<string>(nameof(ScriptContent), ["properties", "scriptContent"]);
+        _azCliVersion = DefineProperty<string>(nameof(AZCliVersion), ["properties", "azCliVersion"]);
+        _azPowerShellVersion = DefineProperty<string>(nameof(AZPowerShellVersion), ["properties", "azPowerShellVersion"]);
+        _retentionInterval = DefineProperty<TimeSpan>(nameof(RetentionInterval), ["properties", "retentionInterval"], format: "P");
+        _environmentVariables = DefineListProperty<ContainerAppEnvironmentVariable>(nameof(EnvironmentVariables), ["properties", "environmentVariables"]);
```

Comment:
Just define it and avoid the dep.

Reply:
I think I tried at first ... went to the rabbit hole. Here is what it takes to serialize in Bicep, and it has private dependencies AFAIR. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerapps/Azure.ResourceManager.AppContainers/src/Generated/Models/ContainerAppEnvironmentVariable.Serialization.cs

Reply:
Maybe not actually: [ContainerAppEnvironmentVariable](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/provisioning/Azure.Provisioning.AppContainers/src/Generated/Models/ContainerAppEnvironmentVariable.cs)

Will try again